mongodb/vibe/db/mongo/sasl.d

```diff
@@ -36,17 +37,22 @@ package struct ScramState
 	private DigestType!SHA1 m_saltedPassword;
 	private string m_authMessage;
 
-	string createInitialRequest(string user)
+	string createInitialRequest(string user, string nonce = null)
 	{
-		ubyte[18] randomBytes;
-		g_rng.read(randomBytes[]);
-		m_nonce = Base64.encode(randomBytes);
+		if (nonce.length) {
+			m_nonce = nonce;
```

Comment:
Where is this code called from?
I see a unittest, but that's it?

Reply:
the optional nonce is just for the unittest so we can actually put in predictable values and what they should result in

Reply:
made this a `version (unittest)` block